### ISSUE TITLE
Splitting up the fastfile

### ISSUE THREAD

**Issue by Alarson93** [fastlane/fastlane#13929]
### Question Checklist

- [ ] Updated fastlane to the latest version
- [ ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [ ] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

I've been working on a Fastfile that will automate deployment of Xamarin iOS projects. We have many different configurations: Internal Build, Internal Testflight, External Staging, External Release, Enterprise, etc.

Everything looks A-OK. The builds have all worked and uploaded correctly. 

The problem is the size of our Fastfile. With so many lanes and custom steps (msbuild, configuring signing, modifying plist values) our Fastfile is just shy of 300 lines. While this isn't terrible, it feels like this could be better abstracted.

For instance - I would like to have a file for each lane (where the steps to each lane are implemented) and a "Steps" directory that includes a file for each custom step (such as MSBuild). For the time being I simply made them Ruby files. The lanes execute applicable functions in the ruby files. Ex:
```  lane :internal_build do
    InternalBuild.start
  end```

The problem is that I don't know how to access and Fastlane actions from this other files. Like - how do I call `get_info_plist_value(key: "something")` from one of these Ruby files?

TLDR - I want to break up my Fastfile. This is easy for the most part, but I'm unsure how to call the Fastlane Actions from these other files.

**Comment by janpio** [comment]
I think this might help: https://docs.fastlane.tools/actions/import/#import

**Comment by Alarson93** [comment]
Thank you for your reply @janpio, but I think that is the opposite of what I want to do.

I want a Ruby file (MyFile.rb) to be able to call `get_info_plist_value` from my Fastfile.
The link you point me to would be importing another Fastfile into my main Fastfile. 

I suppose I _could_ make all the other files Fastfiles instead of Ruby files, it just feels a bit strange. I'll try that if I have time this afternoon.

**Comment by janpio** [comment]
Ah, I thought the ruby files were a workaround for your wish to be able to extract all the lanes and steps into their individual files - which would be possible with `import`.

I would try, but am not sure if or which will work, `Fastlane::Actions::PromptAction()` or `other_action.prompt` for e.g. the `prompt` action. Let's hope someone replies that actually knows what they are doing 🙈

**Comment by janpio** [comment]
> Interestingly - once F imports all the files....the other files can call each other without importing one another. For instance: A can call `build`, which is on C, without importing C.
> (It seems that Fastlane shares imports between files???)

`import` is effectively opening the file, reading its content and `eval`ing it, it seems:
https://github.com/fastlane/fastlane/blob/87864cf27db18e60eefeb670a68f5194596d5019/fastlane/lib/fastlane/fast_file.rb#L212-L233
https://github.com/fastlane/fastlane/blob/87864cf27db18e60eefeb670a68f5194596d5019/fastlane/lib/fastlane/fast_file.rb#L42-L71

So you could/should probably just move the `import` to the top of your file as a form of `include` - the content is all "executed" in the context of the main `Fastfile`.
